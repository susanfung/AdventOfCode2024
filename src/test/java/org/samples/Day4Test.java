package org.samples;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.IOException;

public class Day4Test {
    @Test
    public void testDay4Part1_findXmasWithMarkers() throws IOException {
        Day4 day4 = new Day4();
        String filePath = "src/test/java/org/samples/Day4TestData.txt";

        String result = day4.findXmasWithMarkers(filePath);

        Approvals.verify(result);
    }

    @Test
    public void testDay4Part1_answer() throws IOException {
        Day4 day4 = new Day4();
        String filePath = "src/test/java/org/samples/Day4Data.txt";

        String result = day4.findXmasWithMarkers(filePath);

        Approvals.verify(result);
    }

    @Test
    public void testDay4Part2_findXMas() throws IOException {
        Day4 day4 = new Day4();
        String filePath = "src/test/java/org/samples/Day4TestData.txt";

        String result = day4.findXMas(filePath);

        Approvals.verify(result);
    }
}
