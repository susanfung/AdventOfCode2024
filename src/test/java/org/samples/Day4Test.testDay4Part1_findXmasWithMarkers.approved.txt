Found 18 XMAS words:

....12222.
.333318...
...A..8...
..9.A.81.5
69677A8.54
9.....A5.4
C.F.B.I.GH
.C.F.D.I.H
..C.F.G.IH
.B.D.GEEEI
